deps/graph-parser/src/logseq/graph_parser/util/db.cljs

```diff
@@ -22,56 +22,93 @@ it will return 1622433600000, which is equivalent to Mon May 31 2021 00 :00:00."
   (parse-long
    (string/replace (date-time-util/ymd date) "/" "")))
 
+(defn old->new-relative-date-format [input]
+  (let [count (re-find #"^\d+" (name input))
+        plus-minus (if (re-find #"after" (name input)) "+" "-")
+        ms? (string/ends-with? (name input) "-ms")]
+    (keyword :today (str plus-minus count "d" (if ms? "-ms" "")))))
+
+(comment
+  (old->new-relative-date-format "1d")
+  (old->new-relative-date-format "1d-before")
+  (old->new-relative-date-format "1d-after")
+  (old->new-relative-date-format "1d-before-ms")
+  (old->new-relative-date-format "1d-after-ms")
+  (old->new-relative-date-format "1w-after-ms"))
+
+(defn keyword-input-dispatch [input]
+  (cond 
+    (#{:current-page :current-block :parent-block :today :yesterday :tomorrow :right-now-ms} input) input
+    (re-find #"^[+-]\d+[dwmy](-ms)?$" (name input)) :relative-date
+    (re-find #"^\d+d(-before|-after|-before-ms|-after-ms)?$" (name input)) :DEPRECATED-relative-date
+    (= :start-of-today-ms input) :DEPRECATED-start-of-today-ms 
+    (= :end-of-today-ms input) :DEPRECATED-end-of-today-ms))
+(defmulti resolve-keyword-input (fn [_db input _opts] (keyword-input-dispatch input))) 
+
+(defmethod resolve-keyword-input :DEPRECATED-relative-date [db input opts]
+  ;; This handles all of the cases covered by the following:
+  ;; :Xd, :Xd-before, :Xd-before-ms, :Xd-after, :Xd-after-ms
+  (resolve-keyword-input db (old->new-relative-date-format input) opts))
+
+(defmethod resolve-keyword-input :DEPRECATED-start-of-today-ms [db _ opts]
+  (resolve-keyword-input db :today/-0d-ms opts))
+
+(defmethod resolve-keyword-input :DEPRECATED-end-of-today-ms [db _ opts] 
+  (resolve-keyword-input db :today/+0d-ms opts))
+
+(defmethod resolve-keyword-input :current-page [_ _ {:keys [current-page-fn]}]
+  ;; TODO: handle current-page-fn not being provided
+  (some-> (current-page-fn) string/lower-case))
+
+(defmethod resolve-keyword-input :current-block [db _ {:keys [current-block-uuid]}]
+  ;; TODO: handle current-block-uuid not being provided
+  (:db/id (d/entity db [:block/uuid current-block-uuid])))
+
+(defmethod resolve-keyword-input :parent-block [db _ {:keys [current-block-uuid]}]
+  ;; TODO: handle current-block-uuid not being provided
+  (:db/id (:block/parent (d/entity db [:block/uuid current-block-uuid]))))
+
+(defmethod resolve-keyword-input :today [_ _ _]
+  (date->int (t/today)))
+
+(defmethod resolve-keyword-input :yesterday [_ _ _]
+  (date->int (t/minus (t/today) (t/days 1))))
+
+(defmethod resolve-keyword-input :tomorrow [_ _ _]
+  (date->int (t/plus (t/today) (t/days 1))))
+
+(defmethod resolve-keyword-input :right-now-ms [_ _ _]
+  (date-time-util/time-ms))
+
+(defmethod resolve-keyword-input :relative-date [db input opts]
+  (let [relative-to (case (or (namespace input) "today")
```

Comment:
This case looks strange now, but shortly will add a `"journal"` case, and in the future could add things like `"prop-.*"` case to have queries be relative to a block property 

so something like the following would resolve relative to 20230101, and you could change the prop instead of mucking with the query:

```
date:: 20230101
#+BEGIN_QUERY 
{:name "test"
 :inputs [:prop-date/-3d :prop-date/+3d]
 :query [:find (pull ?b [*])
         :in $ ?before ?after 
         (between ?b ?before ?after)]}
#+END_QUERY
```

Reply:
Could also actually support `query` and `current` as I originally anticipated - if people are embedded blocks across journal pages ¯\_(ツ)_/¯

Reply:
case makes sense for now. I'm not sure if properties as part of the input syntax would be easy to support as properties can have namespaces and pretty much anything that is valid edn including whitespace

Reply:
I think full support for properties then may be a challenge - but to me it would be satisfactory if it only did a match on the name, and would fail if there was anything besides a date type object in the value